Warn about carefully placing the --self-contained flag

When placed at the end of the options, the `--self-contained` flag will not work:

    dotnet publish --runtime win-x64 --configuration debug --self-contained project.fsproj
    [...]
    C:\Program Files\dotnet\sdk\2.2.203\Sdks\Microsoft.NET.Sdk\target\Microsoft.NET.Sdk.targets(146,23): error MSB4030: "project.fsproj" is an invalid value for the "IsSelfContained" parameter of the "GenerateDepsFile" task. The "IsSelfContained" parameter is of type "System.Boolean".

The fix is either to supply `true` to `--self-contained`:

    dotnet publish --runtime win-x64 --configuration debug --self-contained true project.fsproj

or to rearrange the options:

    dotnet publish --self-contained --runtime win-x64 --configuration debug project.fsproj





---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 211afdc7-c796-202d-239d-823f2a499dff
* Version Independent ID: 9439897e-dcb3-bc2e-5cae-35c1ca2e0681
* Content: [dotnet publish command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore21)
* Content Source: [docs/core/tools/dotnet-publish.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-publish.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -32,7 +32,7 @@ dotnet publish [-h|--help]
 - A *.runtimeconfig.json* file that specifies the shared runtime that the application expects, as well as other configuration options for the runtime (for example, garbage collection type).
 - The application's dependencies, which are copied from the NuGet cache into the output folder.
 
-The `dotnet publish` command's output is ready for deployment to a hosting system (for example, a server, PC, Mac, laptop) for execution. It's the only officially supported way to prepare the application for deployment. Depending on the type of deployment that the project specifies, the hosting system may or may not have the .NET Core shared runtime installed on it.
+The `dotnet publish` command's output is ready for deployment to a hosting system (for example, a server, PC, Mac, laptop) for execution. It's the only officially supported way to prepare the application for deployment. Depending on the type of deployment that the project specifies, the hosting system may or may not have the .NET Core shared runtime installed on it. For more information, see [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
 
 ## Arguments
 

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -108,6 +108,8 @@ The `dotnet publish` command's output is ready for deployment to a hosting syste
 
   Publishes the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Default is `true` if a runtime identifier is specified. For more information, see [.NET Core application publishing](../deploying/index.md) and [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
 
+  If this option is used without specifying `true` or `false`, the default is `true`. In that case, don't put the solution or project argument immediately after `--self-contained`, because `true` or `false` is expected in that position.
+
 - **`--no-self-contained`**
 
   Equivalent to `--self-contained false`. Available since .NET Core 3.0 SDK.

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -106,7 +106,7 @@ The `dotnet publish` command's output is ready for deployment to a hosting syste
 
 - **`--self-contained [true|false]`**
 
-  Publishes the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Default is `true` if a runtime identifier is specified and the project isn't a library project. For more information, see [.NET Core application publishing](../deploying/index.md) and [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
+  Publishes the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Default is `true` if a runtime identifier is specified and the project is an executable project (not a library project). For more information, see [.NET Core application publishing](../deploying/index.md) and [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
 
   If this option is used without specifying `true` or `false`, the default is `true`. In that case, don't put the solution or project argument immediately after `--self-contained`, because `true` or `false` is expected in that position.
 

--- a/docs/core/tools/dotnet-publish.md
+++ b/docs/core/tools/dotnet-publish.md
@@ -106,7 +106,7 @@ The `dotnet publish` command's output is ready for deployment to a hosting syste
 
 - **`--self-contained [true|false]`**
 
-  Publishes the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Default is `true` if a runtime identifier is specified. For more information, see [.NET Core application publishing](../deploying/index.md) and [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
+  Publishes the .NET Core runtime with your application so the runtime doesn't need to be installed on the target machine. Default is `true` if a runtime identifier is specified and the project isn't a library project. For more information, see [.NET Core application publishing](../deploying/index.md) and [Publish .NET Core apps with the .NET Core CLI](../deploying/deploy-with-cli.md).
 
   If this option is used without specifying `true` or `false`, the default is `true`. In that case, don't put the solution or project argument immediately after `--self-contained`, because `true` or `false` is expected in that position.
 

